tee-worker/ts-tests/integration-tests/common/utils/assertion.ts

```diff
@@ -276,35 +279,42 @@ export async function assertVc(context: IntegrationTestContext, subject: CorePri
     const { proof, ...vcWithoutProof } = vcPayloadJson;
 
     // step 5
-    // prepare teerex enclave registry data for further checks
+    // prepare teebag enclave registry data for further checks
     const parachainBlockHash = await context.api.query.system.blockHash(vcPayloadJson.parachainBlockNumber);
     const apiAtVcIssuedBlock = await context.api.at(parachainBlockHash);
-    const enclaveCount = await apiAtVcIssuedBlock.query.teerex.enclaveCount();
-
-    const lastRegisteredEnclave = (await apiAtVcIssuedBlock.query.teerex.enclaveRegistry(enclaveCount))
-        .value as TeerexPrimitivesEnclave;
+    const enclaveIdentifier = context.api.createType(
+        'Vec<AccountId>',
+        await apiAtVcIssuedBlock.query.teebag.enclaveIdentifier('Identity')
+    );
+    const lastRegisteredEnclave = (
+        await apiAtVcIssuedBlock.query.teebag.enclaveRegistry(enclaveIdentifier[enclaveIdentifier.length - 1])
+    ).toHuman() as unknown as PalletTeebagEnclave;
 
     // step 6
     // check vc signature
     const signature = Buffer.from(hexToU8a(`0x${proof.proofValue}`));
     const message = Buffer.from(JSON.stringify(vcWithoutProof));
-    const vcPubkey = Buffer.from(lastRegisteredEnclave.vcPubkey.value);
+    const vcPubkeyBytes = context.api.createType('Option<Bytes>', lastRegisteredEnclave.vcPubkey).unwrap();
+    const vcPubkey = Buffer.from(hexToU8a(vcPubkeyBytes.toHex()));
     const signatureStatus = await ed.verify(signature, message, vcPubkey);
 
     assert.isTrue(signatureStatus, 'Check Vc signature error: signature should be valid');
 
     // step 7
     // check VC mrenclave with enclave's mrenclave from registry
+
+    // this is awful, but I have no idea why base58.encode(registryMrenclave.mrenclave) doesn't work - it's a U8aFixed
+    const registryMrenclave = context.api.createType('[u8; 32]', lastRegisteredEnclave.mrenclave);
```

Comment:
If you have an idea why `base58.encode(registryMrenclave.mrenclave)` doesn't work - please let me know

@Traf333 @jonalvarezz @Verin1005 @grumpygreenguy

Reply:
can't tell exactly, I'd need to toy around it, but the `.toHuman() as PalletTeebagEnclave` of L291 might be encoding it in a different format? the `toHuman` output isn't exactly `PalletTeebagEnclave` – `api.createType` should be more accurate.

Reply:
Thanks @jonalvarezz - thanks seems to be the cause!

Although I haven't understood it comprehensively, but the `teebag.enclaveRegistry` query returns an `Option<_>`, so maybe `toHuman()` then `as PalletTeebagEnclave` will pull in some tricky decoding problem which silently gets ignored.

Either way, I just updated it - it seems to be working now :)
https://github.com/litentry/litentry-parachain/pull/2469/commits/a00123ba0f3d7e5e2f8698cb81400a58c1253a11

Lesson learned is be cautious with using `as unknown as` and `toHuman`

Reply:
Yeah, `toHuman` and `toJson` should ideally be used for displaying or logging purposes only. We've found ourselves in situations where we had to use it when an `api`/ `registry` instance was not worth creating.